### ISSUE TITLE
False positive InvalidArgument for count for callable-array

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2012]
```php
<?php

class MyClass {
    public function __construct($x) {
        $this->callback = $x;
    }

    /** @var callable|false */
    public $callback;
    function example() : void {
        if (is_array($this->callback)) {
            // Unexpected ERROR: InvalidArgument - Argument 1 of count expects array<array-key, mixed>|Countable|SimpleXMLElement|ResourceBundle, callable-array{0: string|object, 1: string} provided
            echo "Count is: " . count($this->callback);
        }
    }
}
```

**Comment by muglug** [comment]
I sort of get that it's a bug, but won't the result always be 2? I think I should make `int(2)` the return type when calling `count` on a `callable-array`

**Comment by TysonAndre** [comment]
Yes, it makes sense to always be 2 if callable-array is the only array/countable(or similar) type.

I had code where the union type of $this->callback really should have been `callable|array{0:string}|false`
